### ISSUE TITLE
npm install vulnerabilities @4.0.1

### ISSUE THREAD

**Issue by dewwwald** [watson-developer-cloud/node-sdk#857]
Remember, an issue is not the place to ask questions. You can use [Stack Overflow](http://stackoverflow.com/questions/tagged/ibm-watson) for that, or you may want to start a discussion on the [dW Answers](https://developer.ibm.com/answers/questions/ask/?topics=watson).

 Before you open an issue, please check if a similar issue already exists or has been closed before.

**Check service status**

1.  For service issues or 5xx errors, first, go to https://console.bluemix.net/status?tags=platform,runtimes,services&view=n:i and check the status of the service.
1.  If the service status is OK, continue with a bug report.

---

**Overview**
As of 11 April 2019 running npm install gives a large amount of security issues.

**Expected behavior**
When running an install of a package like this one does not expect vulnerabilities. However...

**Actual behavior**
After running an `npm install --save watson-developer-cloud` as described by the documentation I get the following message from npm.

```
found 21 vulnerabilities (3 low, 14 moderate, 4 high) in 27607 scanned packages
  run `npm audit fix` to fix 10 of them.
  11 vulnerabilities require manual review. See the full report for details.
```

**How to reproduce**
To reproduce this start a new node project and install watson developer cloud as outlined in the first few lines of this document. https://cloud.ibm.com/apidocs/natural-language-understanding?code=node

**Screenshots**
_If applicable, add screenshots to help explain your problem._

**SDK Version**
Version 2.29.0

**Additional information:**
- OS: IOS
- Which version of `Node` are you using? 10.15.3

**Additional context**
Here is `npm audit --parseable | awk -F $'\t' '{print $1,$2,$4,$5,$6}'` command output.
review string No patch available Regular Expression Denial of Service https://npmjs.com/advisories/536
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review request >=2.68.0 Remote Memory Exposure https://npmjs.com/advisories/309
review tunnel-agent >=0.6.0 Memory Exposure https://npmjs.com/advisories/598

Happy to help in any way I can.

**Comment by dpopp07** [comment]
Hey @dewwwald 

I think this is the problem:

> Version 2.29.0

We are currently on Version 4.0.1 and v2 is no longer supported for security issues. To avoid these insecure dependencies, I strongly recommend you update to the latest version or at the very least v3.18.4.

Also, if you do decide to use the latest version, you should use the package `ibm-watson`, i.e. `npm install ibm-watson`. We moved the package there from `watson-developer-cloud` and it will be receiving all future updates.

**Comment by dewwwald** [comment]
@dpopp07  That is good to know. Thank you for that. 

I am sorry that I did not check the version. I am also not sure why the installation decided to install that version (I don't have it in my package.json).

Another note, the documentation does not say to install `ibm-watson` instead.

After installing `ibm-watson@4.0.1` (which is the latest), I get the following:
found 12 vulnerabilities (11 moderate, 1 high)

review string No patch available Regular Expression Denial of Service https://npmjs.com/advisories/536
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review hoek > 4.2.0 < 5.0.0 || >= 5.0.3 Prototype Pollution https://npmjs.com/advisories/566
review request >=2.68.0 Remote Memory Exposure https://npmjs.com/advisories/309
review tunnel-agent >=0.6.0 Memory Exposure https://npmjs.com/advisories/598
review marked >=0.6.2 Regular Expression Denial of Service https://npmjs.com/advisories/812

**Comment by dewwwald** [comment]
Sorry about the previous error report. This is not an issue with ibm-watson package but rather my repo. The previous developer instlalled the package, seemingly manually, and then added 'file:watson-developer-cloud' to the package. Pointing to the root of the project at watson-developer-cloud Was this a recommended way to do this before? I am not familiar with this approach.

Now I only get this error message: `npm audit --parseable | awk -F $'\t' '{print $1,$2,$4,$5,$6}'`
review marked >=0.6.2 Regular Expression Denial of Service https://npmjs.com/advisories/812

**Comment by dpopp07** [comment]
@dewwwald No problem! That is really bizarre, I highly doubt that was a recommended pattern but I also wasn't working on this project at the time. The [install command in the readme](https://github.com/watson-developer-cloud/node-sdk#installation) is updated but it is probably a good idea to add a message about the transition in the readme as well.

Installing `watson-developer-cloud` should also give a deprecation warning but might not have for you since it was embedded.

When I run the audit, I get a few vulnerabilities so I'll update the dependencies. Thanks for the heads up

**Comment by dewwwald** [comment]
Updating the subject line to be more searchable.

**Comment by dpopp07** [comment]
Resolved with #858
